src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -2408,16 +2408,18 @@ .maxstack  8
 
             diff7.VerifyIL(
 @"{
-  // Code size       16 (0x10)
-  .maxstack  8
+  // Code size       18 (0x12)
+  .maxstack  1
   IL_0000:  nop
   IL_0001:  ldarg.0
   IL_0002:  call       0x06000008
   IL_0007:  nop
   IL_0008:  ldarg.0
-  IL_0009:  call       0x06000009
-  IL_000e:  nop
-  IL_000f:  ret
+  IL_0009:  stloc.0
```

Comment:
Why do we need an explicit cast here when we did not before?

Reply:
I think this is because of a fix for https://github.com/dotnet/roslyn/issues/22858. I forgot to mention that this was also fixed.

Reply:
Yes, this is because of a fix for [https://github.com/dotnet/roslyn/issues/22858](https://github.com/dotnet/roslyn/issues/22858). I forgot to mention that this was also fixed.

Covariantly compatible arrays are "assignment-compatible" but not completely substitutable. Fetching `ref Base` from a `Derived[]` instance is unverifiable. 
Since we do not know what operations will be applied, we cannot drop implicit casts to array types or we are risking unverifiable code. 
We had a minor regression from the native compiler here.

---
In reply to: [149520969](https://github.com/dotnet/roslyn/pull/23055#discussion_r149520969) [](ancestors = 149520969)